Dockerfile

```diff
@@ -0,0 +1,76 @@
+### Build stage ###
+FROM golang:1.12-alpine AS build
+
+RUN \
+echo "**** install build dependencies ****" && \
+apk add --no-cache gcc git musl-dev && \
+mkdir /app
+
+WORKDIR /app
+
+# When building from the gohub repository, just copy it in
+copy . .
```

Comment:
It may make sense to split the build into two parts: first get dependencies, then build the actual source. It should speed up consequent Docker image rebuilds:

```Dockerfile
# Copy go.mod and go.sum and get all the dependecies
COPY go.* ./ 
RUN go get -d -v ./...

# Build
COPY . .
# ... the rest of the file
```